### PR TITLE
SUPESC-492: Standalone testing

### PR DESCRIPTION
**Pull request by luninaleksey**

### CODE REVIEW

**Comment by dereuromark** [comment]
Looks like the php7.4 issue of inside php8 compatibility is still here
Which I cannot reproduce locally.
Can you reproduce it?

File: phpstan.neon [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
     checkGenericClassInNonGenericObjectType: false
     checkMissingIterableValueType: false
     ignoreErrors:
-        - '#.+Generated\\Shared\\Transfer.+#'
-        - '#Parameter .+ of method .+ has invalid typehint type Orm\\Zed\\Payone\\Persistence\\.+#'
-        - '#Return typehint of method .+ has invalid type Orm\\Zed\\Payone\\Persistence\\.+#'
+        - '#Expression on left side of \?\? is not nullable\.#'
+        - '#Parameter .+ of method .+ has invalid type .+#'
+        - '#Method .+ has invalid return type .+#'
+        - '#Method .+ should return .+ but returns .+#'
```
Comment (dereuromark) [inline phpstan.neon:13]: Those are too generic, they match everything and are not valid as such.

File: src/SprykerEco/Client/Payone/ClientApi/Request/AbstractRequest.php [hunk 2]
```diff
@@ -230,7 +230,7 @@ public function getResponseType(): ?string
      *
      * @return void
      */
-    public function setIntegratorName($integrator_name): void
+    public function setIntegratorName(string $integrator_name): void
```
Comment (dereuromark) [inline src/SprykerEco/Client/Payone/ClientApi/Request/AbstractRequest.php:233]:
```suggestion
    public function setIntegratorName(string $integratorName): void
```

File: ci/config_propel.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                'object' => ObjectBuilder::class,
+                'objectstub' => ExtensionObjectBuilder::class,
+                'query' => QueryBuilder::class,
+                'querystub' => ExtensionQueryBuilder::class,
+            ],
+        ],
+    ],
+    'paths' => [
+        'phpDir' => APPLICATION_ROOT_DIR,
+        'sqlDir' => APPLICATION_ROOT_DIR . 'src/Orm/Propel/Sql',
```
Comment (dereuromark) [inline ci/config_propel.php:84]: This and all following should always have trailing slash

File: composer.json [hunk 4]
```diff
@@ -27,7 +28,7 @@
         "spryker/zed-request": "^3.2.0"
     },
     "require-dev": {
-        "phpstan/phpstan": "*",
+        "phpstan/phpstan": "^0.12.57",
```
Comment (dereuromark) [inline composer.json:31]: 1.0.0 !

Comment (luninaleksey) [reply composer.json]: Fixed.

File: ci/config_propel.php (hunk 3 above)
Comment (luninaleksey) [reply ci/config_propel.php]: Fixed.

File: src/SprykerEco/Client/Payone/ClientApi/Request/AbstractRequest.php (hunk 2 above)
Comment (luninaleksey) [reply src/SprykerEco/Client/Payone/ClientApi/Request/AbstractRequest.php]: Fixed.

File: phpstan.neon (hunk 1 above)
Comment (luninaleksey) [reply phpstan.neon]: Fixed.

File: composer.json [hunk 5]
```diff
@@ -6,6 +6,8 @@
     "require": {
         "php": ">=7.4",
         "spryker/calculation": "^4.0.0",
+        "symfony/event-dispatcher-contracts": "^v2.5.0",
+        "symfony/string": "^v5.4.0",
```
Comment (dereuromark) [inline composer.json:10]: We never include v in versions, those are not needed.

Comment (luninaleksey) [reply composer.json]: Fixed.

**Comment by luninaleksey** [comment]
Problem with 7.4 is fixed. As a result we have a problem with 7.4 and 8.0 both. It's
`In ConnectionFactory.php line 42:
------------------------------------------------------------
  Unable to open connection  
In PdoAdapter.php line 82:
  Unable to open PDO connection  
In PdoConnection.php line 82:
  SQLSTATE[HY000] [2002] Connection refused  `

**Comment by dereuromark** [comment]
"As a result"
That is not true.
This was already there before, the question is now can we connect here to the DB?
How does this work locally for you? Try to reproduce exact steps and see if you can connect to a test DB on your machine given similar config added or alike.
This might be easier than trying to get CI to run via blind commits.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -1,124 +1,124 @@
...
+    DATABASE_ALLOW_EMPTY_PASSWORD: false
+    DATABASE_CHARACTER_SET: utf8
+    DATABASE_COLLATE: utf8_general_ci
+
 on:
-    pull_request:
     push:
         branches:
-            - master
-    workflow_dispatch:
```
Comment (dereuromark) [inline .github/workflows/ci.yml:8]: What are these kind of noise and not OK changes?

File: ci/stores.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        ]
+    ],
+    'locales' => [
+        'de' => 'de_DE',
+    ],
+    'countries' => [
+        'DE',
+        'AT',
+        'NO',
+        'CH',
```
Comment (dereuromark) [inline ci/stores.php:18]: Do we need more than DE/GB for testing?

**Comment by dereuromark** [comment]
Your CI changes are invalid and effectively removed all CI.

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -1,124 +1,159 @@
...
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-version }}
+                  extensions: mbstring, intl, pdo_mysql
+
+            - name: Composer prefer lowest
+              run: composer lowest-setup
+
+            - name: Run PHPStan (currently not running properly)
+              run: composer stan || true
```
Comment (dereuromark) [inline .github/workflows/ci.yml:156]: do we need the true hack? or does it run here fine?

File: phpstan.neon [hunk 9]
```diff
@@ -1,22 +1,26 @@
...
-        - '#Parameter .+ of method .+ has invalid typehint type Orm\\Zed\\Payone\\Persistence\\.+#'
-        - '#Return typehint of method .+ has invalid type Orm\\Zed\\Payone\\Persistence\\.+#'
-        - '#Call to static method create\(\) on an unknown class Orm\\Zed\\.+\\Persistence\\.+#'
-        - '#Instantiated class Orm\\Zed\\Payone\\Persistence\\.+ not found.+#'
-        - '#Parameter .+ of method .+ has invalid typehint type Orm\\Zed\\Sales\\Persistence\\.+#'
-        - '#Call to method .+ on an unknown class Orm\\Zed\\.+\\Persistence\\.+#'
-        - '#PHPDoc tag @var for variable .+ contains unknown class Orm\\Zed\\.+\\Persistence\\.+#'
+        - '#Expression on left side of \?\? is not nullable\.#'
+        - '#PHPDoc type .+ of property .+ is not covariant with PHPDoc type .+ of overridden property .+#'
+        - '#Property .+ does not accept .+#'
```
Comment (dereuromark) [inline phpstan.neon:12]: These are still to broad IMO.

File: .github/workflows/ci.yml (hunk 8 above)
Comment (luninaleksey) [reply .github/workflows/ci.yml:156]: Unfortunately, no. We still need it.

File: phpstan.neon (hunk 9 above)
Comment (luninaleksey) [reply phpstan.neon:12]: I can work on this place some more if it should.

Comment (luninaleksey) [reply phpstan.neon]: Fixed.

File: phpstan.neon [hunk 10]
```diff
@@ -9,11 +9,9 @@ parameters:
     ignoreErrors:
         - '#Expression on left side of \?\? is not nullable\.#'
```
Comment (dereuromark) [inline phpstan.neon:10]: This also looks too broad.

File: phpstan.neon [hunk 11]
```diff
@@ -9,11 +9,9 @@ parameters:
...
         - '#PHPDoc type .+ of property .+ is not covariant with PHPDoc type .+ of overridden property .+#'
```
Comment (dereuromark) [inline phpstan.neon:11]: What causes this?

File: phpstan.neon [hunk 12]
```diff
@@ -9,11 +9,9 @@ parameters:
...
-        - '#Property .+ does not accept .+#'
         - '#Property .+ in .+ is not nullable#'
         - '#Strict comparison using === between string and null will always evaluate to false\.#'
         - '#Property .+ is never read, only written\.#'
-        - '#Trying to invoke .+ but it might not be a callable\.#'
         - '#Parameter .+ of .+ expects .+ given\.#'
         - '#Cannot call method .+ on .+#'
         - '#Variable .+ in .+ is never defined.#'
```
Comment (dereuromark) [inline phpstan.neon:17]: This also looks too broad.

File: phpstan.neon [hunk 13]
```diff
@@ -9,11 +9,9 @@ parameters:
     ignoreErrors:
         - '#Expression on left side of \?\? is not nullable\.#'
         - '#PHPDoc type .+ of property .+ is not covariant with PHPDoc type .+ of overridden property .+#'
-        - '#Property .+ does not accept .+#'
         - '#Property .+ in .+ is not nullable#'
         - '#Strict comparison using === between string and null will always evaluate to false\.#'
         - '#Property .+ is never read, only written\.#'
-        - '#Trying to invoke .+ but it might not be a callable\.#'
         - '#Parameter .+ of .+ expects .+ given\.#'
         - '#Cannot call method .+ on .+#'
```
Comment (dereuromark) [inline phpstan.neon:16]: This also looks too broad.

File: phpstan.neon [hunk 14]
```diff
@@ -9,11 +9,9 @@ parameters:
     ignoreErrors:
         - '#Expression on left side of \?\? is not nullable\.#'
         - '#PHPDoc type .+ of property .+ is not covariant with PHPDoc type .+ of overridden property .+#'
-        - '#Property .+ does not accept .+#'
         - '#Property .+ in .+ is not nullable#'
         - '#Strict comparison using === between string and null will always evaluate to false\.#'
         - '#Property .+ is never read, only written\.#'
-        - '#Trying to invoke .+ but it might not be a callable\.#'
         - '#Parameter .+ of .+ expects .+ given\.#'
```
Comment (dereuromark) [inline phpstan.neon:15]: This also looks too broad.

File: phpstan.neon [hunk 15]
```diff
@@ -9,11 +9,9 @@ parameters:
     ignoreErrors:
         - '#Expression on left side of \?\? is not nullable\.#'
         - '#PHPDoc type .+ of property .+ is not covariant with PHPDoc type .+ of overridden property .+#'
-        - '#Property .+ does not accept .+#'
         - '#Property .+ in .+ is not nullable#'
         - '#Strict comparison using === between string and null will always evaluate to false\.#'
         - '#Property .+ is never read, only written\.#'
```
Comment (dereuromark) [inline phpstan.neon:14]: Shouldnt we clean out those instead?

File: phpstan.neon (hunk 10 above)
Comment (dereuromark) [reply phpstan.neon:10]: Also not needed, as the code can be simplified here

File: phpstan.neon (hunk 15 above)
Comment (luninaleksey) [reply phpstan.neon]: Fixed.

File: phpstan.neon (hunk 14 above)
Comment (luninaleksey) [reply phpstan.neon]: Fixed.

File: phpstan.neon (hunk 13 above)
Comment (luninaleksey) [reply phpstan.neon]: Fixed.

File: phpstan.neon (hunk 12 above)
Comment (luninaleksey) [reply phpstan.neon]: Fixed.

File: phpstan.neon (hunk 11 above)
Comment (luninaleksey) [reply phpstan.neon]: Fixed.

File: phpstan.neon (hunk 10 above)
Comment (luninaleksey) [reply phpstan.neon]: Fixed.

File: composer.json [hunk 16]
```diff
@@ -88,6 +89,12 @@
         "sort-packages": true,
         "platform-check": false
     },
+    "repositories": [
+        {
+            "type": "git",
+            "url": "git@github.com:spryker/spryker.git"
```
Comment (dereuromark) [inline composer.json:95]: wrong repo
you must use the module in question as dependency and using split repo.

**Comment by dereuromark** [comment]
Template for release is missing
Whats the plan here? 
If we dont ship this soon, other PRs cannot be based on top of this, so this PR is blocking right now.

**Comment by luninaleksey** [comment]
This PR will be closed, because this code was merged into https://github.com/spryker-eco/payone/pull/74

**Pull request closed without merging**